### ISSUE TITLE
Panic in handleChunk

### ISSUE THREAD

**Issue by jech** [pions/sctp#17]
I just got me a panic.

    Failed to accept data channel: The association is closed
    panic: close of closed channel

    goroutine 148 [running]:
    github.com/pions/sctp.(*Association).handleChunk(0xc0000a68c0, 0xc000173f40, 0x93ee40, 0xc00012afc0, 0x0, 0x0, 0x0, 0x0, 0x0)
            /home/jch/go/src/github.com/pions/sctp/association.go:769 +0x427
    github.com/pions/sctp.(*Association).handleInbound(0xc0000a68c0, 0xc000396000, 0x30, 0x2000, 0x30, 0x0)
            /home/jch/go/src/github.com/pions/sctp/association.go:239 +0x1c4
    github.com/pions/sctp.(*Association).readLoop(0xc0000a68c0)
            /home/jch/go/src/github.com/pions/sctp/association.go:210 +0xf6
    created by github.com/pions/sctp.Client
            /home/jch/go/src/github.com/pions/sctp/association.go:135 +0x5b

**Comment by enobufs** [comment]
Newer log Hugo reported on Slack:
````
March 25th 2019, 17:17:29.795     -     github.com/pions/sctp.(*Association).handleChunk(0xc000178d10, 0xc000055f40, 0xbb1a40, 0xc000509340, 0x0, 0x0, 0x0, 0x0, 0x0)     - 
    March 25th 2019, 17:17:29.795     -         /go/pkg/mod/github.com/pions/sctp@v1.4.1/association.go:1032 +0x4b9     - 
    March 25th 2019, 17:17:29.795     -     github.com/pions/sctp.(*Association).handleInbound(0xc000178d10, 0xc00051a000, 0x30, 0x2000, 0x30, 0x0)     - 
    March 25th 2019, 17:17:29.795     -         /go/pkg/mod/github.com/pions/sctp@v1.4.1/association.go:256 +0x1c4     - 
    March 25th 2019, 17:17:29.795     -     panic: close of closed channel     - 
    March 25th 2019, 17:17:29.795     -     goroutine 1495 [running]:     - 
    March 25th 2019, 17:17:29.795     -     created by github.com/pions/sctp.Client     - 
    March 25th 2019, 17:17:29.795     -         /go/pkg/mod/github.com/pions/sctp@v1.4.1/association.go:226 +0xf1     - 
    March 25th 2019, 17:17:29.795     -     github.com/pions/sctp.(*Association).readLoop(0xc000178d10)     - 
    March 25th 2019, 17:17:29.795     -         /go/pkg/mod/github.com/pions/sctp@v1.4.1/association.go:141 +0x6f
```

Cookie Echo can be received in Established state when the chunk was retransmitted. We need to check the state to avoid closing the closed channel.

**Comment by enobufs** [comment]
This has been fixed in v1.4.2.
https://github.com/pions/sctp/blob/v1.4.2/association.go#L1105

My notes for SCTP handshake flow and state transmission chart can be found at https://docs.google.com/document/d/18HxKodWkPPfZpKTddQQvepux5MOS3QWMfK1eJYXQdz4
